### PR TITLE
refactor(cache): Restrict type for package cache namespaces

### PR DESCRIPTION
**Pull request by zharinov**
## Changes

This PR makes turns package cache namespaces into restricted set of pre-determined values

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/datasource/git-refs/base.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export abstract class GitDatasource extends Datasource {
   }
 
   @cache({
-    namespace: `datasource-${GitDatasource.id}`,
+    namespace: 'datasource-git',
```
Comment (zharinov) [inline lib/modules/datasource/git-refs/base.ts:25]: Sorry for that, `GitDatasource.id` can't be marked as const because of inheritance

**Comment by rarkins** [comment]
What's the reason to make it strict? e.g. datasource proxy?

**Comment by zharinov** [comment]
Caching layer became too important for treating it as just an optional enhancement. Because of it, I started too look for small steps that could embrace more disciplined usage of it.

File: lib/modules/datasource/git-refs/base.ts (hunk 1 above)
Comment (viceice) [reply lib/modules/datasource/git-refs/base.ts:25]: can you extract the id to a separate const and reuse instead?

File: lib/modules/datasource/terraform-module/base.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ export abstract class TerraformDatasource extends Datasource {
   static id = 'terraform';
 
   @cache({
-    namespace: `datasource-${TerraformDatasource.id}`,
+    namespace: 'datasource-terraform',
```
Comment (viceice) [inline lib/modules/datasource/terraform-module/base.ts:13]: same here

**Review by viceice — APPROVED** [review]

**Review by secustor — APPROVED** [review]

**Pull request merged**